akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1132,6 +1171,12 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
         shutdownSelfWhenDown()
       }
 
+      if (latestGossip.members.exists(m => MembershipState.shutdownStates(m.status)) && (MembershipState
+            .allowedToPrepareToShutdown(selfStatus))) {
+        logInfo("Detected full cluster shutdown")
+        self ! ClusterUserAction.PrepareForShutdown(selfAddress)
```

Comment:
would be more efficient to inline the change, with a self message, because then the latest is sent back immediately `if (talkback)`

Reply:
Ah gotcha. So update latest gossip right away so it is sent out with talkback 👍